targets/lenovo-x1/flake-module.nix

```diff
@@ -3,15 +3,71 @@
 #
 # Configuration for Lenovo X1 Carbon Gen 11
 {
-  inputs,
   lib,
   self,
   ...
 }: let
-  inherit (inputs) microvm lanzaboote disko;
-  name = "lenovo-x1-carbon";
   system = "x86_64-linux";
-  targets = import ./everything.nix {inherit self lib microvm lanzaboote disko name system;};
+
+  mvp-definition = machineType: variant: extraModules: let
+    hostConfiguration = lib.nixosSystem {
+      inherit system;
+      modules =
+        [
+          self.nixosModules.profiles
+          self.nixosModules.laptop
+
+          #TODO can we move microvm to the profile/laptop-x86?
+          self.nixosModules.microvm
+          #TODO see the twisted dependencies in common/desktop
+
+          ({config, ...}: {
+            time.timeZone = "Asia/Dubai";
+
+            ghaf = {
+              profiles = {
+                mvp-user-trial.enable = true;
+
+                laptop-x86 = {
+                  enable = true;
+                  netvmExtraModules = [self.nixosModules.reference-services];
+                  guivmExtraModules = [self.nixosModules.reference-programs];
+                  inherit (config.ghaf.reference.appvms) enabled-app-vms;
+                };
+
+                # variant type, turn on debug or release
+                debug.enable = variant == "debug";
+                release.enable = variant == "release";
+              };
+            };
+          })
+        ]
+        ++ extraModules;
+    };
+  in {
+    inherit hostConfiguration;
+    name = "${machineType}-${variant}";
+    package = hostConfiguration.config.system.build.diskoImages;
+  };
+
+  targets = [
+    (mvp-definition "lenovo-x1-carbon-gen10" "debug" [
+      self.nixosModules.disko-basic-partition-v1
+      {ghaf.hardware.definition.configFile = "/lenovo-x1/definitions/x1-gen10.nix";}
```

Comment:
If we keep defining targets depending on the hardware, we could require that the hardware definition for each `machineType` lives in `target`, and automatically import the right configuration according to the value of `machineType`?  So we have a clean separation between hw definition and the Ghaf hw usage. All that needs to be done downstream is to generate a hw definition file and put it in target, naming it accordingly. But perhaps how we handle the hardwares may be a topic for another time.